### DISCUSSION TITLE
Building a WASM executable

### DISCUSSION THREAD

**Discussion by alephnull** [spf13/viper#1119] [Q&A] [answered]
When trying to build,
```shellsession
% GOOS=js GOARCH=wasm go build -o g.wasm
# github.com/spf13/viper
/home/alok/go/pkg/mod/github.com/spf13/viper@v1.7.0/viper.go:344:19: undefined: fsnotify.NewWatcher
/home/alok/go/pkg/mod/github.com/spf13/viper@v1.7.0/viper.go:366:20: select case must be receive, send or assign recv
/home/alok/go/pkg/mod/github.com/spf13/viper@v1.7.0/viper.go:393:18: select case must be receive, send or assign recv
```

For other packages, it was simple enough to exclude them from wasm builds with build tags but viper is all over the packages I need. Is there a way to use viper for WASM?

**Comment by sagikazarmark** [comment, marked as answer]
I guess we could make the watch functionality non-functional on WASM if it doesn't work anyway. We can do that with some build flags.

Looks like it's been already reported in #1071 (let's track it there)
